python/ray/tune/execution/checkpoint_manager.py

```diff
@@ -55,11 +55,18 @@ def __init__(
             checkpoint_score_order=MIN if checkpoint_score_desc else MAX,
         )
 
+        # NOTE: Determines whether the current memory checkpoint should be preferred
+        # over a persistent checkpoint. An example of when this is used when the
+        # in-memory checkpoint has been set by a scheduler, and the scheduler needs
+        # that specific checkpoint be used for trial restore.
+        self._prefer_memory_checkpoint = False
+
         super().__init__(checkpoint_strategy=checkpoint_strategy, delete_fn=delete_fn)
 
     def handle_checkpoint(self, checkpoint: _TrackedCheckpoint):
         # Set checkpoint ID
-        checkpoint.id = checkpoint.id or self._latest_checkpoint_id
```

Comment:
before and after are the same thing right?

Reply:
It would be resetting checkpoint.id if `checkpoint.id == 0`, since `0` is `False` value.

Reply:
@krfricke

Reply:
This is a good catch, your fix looks correct to me.

I think originally it never was a problem as latest_checkpoint_Id would be 0 so we would just re-set to 0. But this won't be the case anymore after the rest of this PR